frontend/src/components/elements/CollapsibleTable.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  Grid,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TableSortLabel, TextField
+} from "@mui/material";
+import { Row } from "./Row.tsx";
+import { ReactNode, useMemo, useState } from "react";
+interface RowData {
+  value: ReactNode;
+  align?: 'right' | 'left' | 'center';
+}
+
+interface RowProps {
+  id: number;
+  data: RowData[];
+}
+
+interface Column {
+  label: string;
+  align?: 'right' | 'left' | 'center';
+  sortKey?: string;
+}
+
+interface CollapsibleTableProps {
+  columns: Column[];
+  rows: RowProps[];
+  renderExpandableContent?: (row: RowProps) => ReactNode;
+  onActionClick?: (row: RowProps) => void;
+}
+
+export default function CollapsibleTable(
+  {
+    columns,
+    rows,
+    renderExpandableContent,
+    onActionClick,
+  }: CollapsibleTableProps) {
+  const [sortKey, setSortKey] = useState<string | null>(null);
+  const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>('asc');
+  const [filter, setFilter] = useState('');
+
+  const handleSort = (key: string | undefined) => {
+    if (!key) return;
+
+    if (sortKey === key) {
+      setSortDirection((prev) => (prev === 'asc' ? 'desc' : 'asc'));
+    } else {
+      setSortKey(key);
+      setSortDirection('asc');
+    }
+  };
+
+  const filteredRows = useMemo(() => {
+    if (!filter) return rows;
+
```

Comment:
const loweredFilter = filter.toLowerCase() and after use it

To avoid too much useless call